src/lua/api-gateway/cache/store/redisCache.lua

```diff
@@ -163,20 +245,28 @@ function _M:evict(key)
         ngx.log(ngx.WARN, "Could not evict an empty key")
         return
     end
-    local redis_rw = redis:new()
-    local redis_host, redis_port = getRedisUpstream(REDIS_RW_UPSTREAM)
-    local ok, err = redis_rw:connect(redis_host, redis_port)
+
+    local upstreamConfig = self.redis_connection_provider.redis_rw_upstream_connection_data or REDIS_RW_UPSTREAM
+    if self.redis_connection_provider.redis_rw_upstream_connection_data and
+            type(self.redis_connection_provider.redis_rw_upstream_connection_data) == 'table' then
+        upstreamConfig = {
+            upstream = self.redis_connection_provider.redis_rw_upstream_connection_data.upstream,
+            password = os.getenv(self.redis_connection_provider.redis_rw_upstream_connection_data.password)
+        }
```

Comment:
can we please extract as the return of a function or a module variable, so we don't duplicate?